### DISCUSSION TITLE
Is it possible to use the prepare script to download non-SLR12 corpora?

### DISCUSSION THREAD

**Discussion by duhumail** [k2-fsa/icefall#1559] [General]
i am beginner, but this question really confused me .
Because slr12 is mainly an English corpus, is it possible that I can find a way to use other slr ID in the prepare script configuration? Even the corpus I prepared myself. Thank you so much

**Comment by JinZr** [comment]
by slr 12 do you mean the https://openslr.org/12/ LibriSpeech corpus?

if you have looked into the prepare.sh script, you will find out dataset downloading functionality is provided by the lhotse toolkit, if there is already a recipe provided in lhotse then it is ok for you to use off-the-shelf command for downloading the corpus. 

you can check recipes supported by lhotse here: https://github.com/lhotse-speech/lhotse/tree/master/lhotse/recipes

**Comment by duhumail** [comment]
yes. I found lhotse command can set dataset download & prepare dataset.
checked the prepare.sh script part:
  if [ ! -d $dl_dir/LibriSpeech/train-other-500 ]; then
    lhotse download librispeech --full $dl_dir
  fi
AND
  if [ ! -e data/manifests/.librispeech.done ]; then
    lhotse prepare librispeech -j $nj $dl_dir/LibriSpeech data/manifests
    touch data/manifests/.librispeech.done
  fi
So I should replace librispeech  with other dataset ?   
thanks for your reply.
BTW , Is it possbile to upload my private dataset ?

**Comment by JinZr** [comment]
yes

i think you can find instruction on how to share your datasets on openslr.org <http://openslr.org/>
